lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,374 @@
+                            topics: record[0].topics
+                        };
+
+                        if (ddqBackendInstance.currentlyPolling === true) {
+                            ddqBackendInstance.poller = timers.setTimeout(checkNow, ddqBackendInstance.config.pollingDelayMs);
+                        }
+
+                        ddqBackendInstance.emit(EMIT_DATA, wrappedMessage);
+                    }
+                }
+            );
+        }
+
+        ddqBackendInstance.poller = timers.setTimeout(checkNow, ddqBackendInstance.config.pollingDelayMs);
+    }
+
+
+    /**
+     * Ensures that heartbeatCleanup is ready to run according to the
+     * heartbeatCleanupDelay. This will continue until pausePolling is called.
+     *
+     * @param {Object} ddqBackendInstance
+     */
+    function restore(ddqBackendInstance) {
+        /**
+         * Queries for jobs that have stopped, and primes them to be cleaned up.
+         */
+        function heartbeatCleanup() {
+            ddqBackendInstance.restorer = null;
+
+            ddqBackendInstance.connection.query(`UPDATE ??
+                SET isProcessing = false,
+                    owner = null,
+                    requeued = false
+                WHERE DATEDIFF(NOW(), SEC_TO_TIME(?)) > heartbeatDate
+                    AND isProcessing = true;`,
+                [
+                    ddqBackendInstance.config.table,
+                    ddqBackendInstance.config.heartbeatLifetimeSeconds
+                ], (err) => {
+                    if (err) {
+                        ddqBackendInstance.emit(EMIT_ERR, err);
+                    }
+
+                    if (ddqBackendInstance.currentlyRestoring === true) {
+                        ddqBackendInstance.restorer = timers.setTimeout(heartbeatCleanup, ddqBackendInstance.config.heartbeatCleanupDelayMs);
+                    }
+                }
+            );
+        }
+
+        ddqBackendInstance.restorer = timers.setTimeout(heartbeatCleanup, ddqBackendInstance.config.heartbeatCleanupDelayMs);
+    }
+
+
+    /**
+     * Inserts a record to the database. If it's unable to do so because the
+     * record already exists, it will check directly that the entry exists
+     * and if it does, attempt to update the record.
+     *
+     * Inserting a record will repeat as many times as are defined in the
+     * config's createMessageCycleLimit.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {Object} params
+     */
+    function trySendMessage(ddqBackendInstance, params) {
+        if (cycleCounter < ddqBackendInstance.config.createMessageCycleLimit) {
+            cycleCounter += 1;
+            ddqBackendInstance.connection.query(`INSERT INTO ?? (hash, messageBase64, topics)
+                VALUES (?, ?, ?);`,
+                [
+                    ddqBackendInstance.config.table,
+                    params.hash,
+                    params.message,
+                    params.topics
```

Comment:
I think we're on the same page now. After reviewing the documentation for IN, it looks like I misunderstood how it handles things. I'll likely need your help or Kyle's help supporting multiple topics, so, for now, I'm going to convert this to a single topic.